File: app/src/main/java/com/BlizzardArmory/warcraft/Talents/SpecializationData/Role.java

package com.BlizzardArmory.warcraft.Talents.SpecializationData;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

public class Role {

    @SerializedName("type")
    @Expose
    private String type;
    @SerializedName("name")
    @Expose
    private String name;

    /**
     * No args constructor for use in serialization
     */
    public Role() {
    }

    /**
     * @param name
     * @param type
     */
    public Role(String type, String name) {
        super();
        this.type = type;
        this.name = name;
    }

    public String getType() {
        return type;
    }

    public void setType(String type) {
        this.type = type;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

}

File: app/src/main/java/com/BlizzardArmory/model/warcraft/talents/specializationdata/Role.kt
package com.BlizzardArmory.model.warcraft.talents.specializationdata

import com.google.gson.annotations.SerializedName


/**
 * The type Role.
 */
data class Role(

    @SerializedName("type")
    var type: String,

    @SerializedName("name")
    var name: String
)